File: packages/python/src/armonik/client/health_checks.py
from grpc import Channel

from ..protogen.client.health_checks_service_pb2_grpc import HealthChecksServiceStub
from ..protogen.common.health_checks_common_pb2 import (
    CheckHealthRequest,
    CheckHealthResponse,
)


class ArmoniKHealthChecks:
    def __init__(self, grpc_channel: Channel):
        """Result service client

        Args:
            grpc_channel: gRPC channel to use
        """
        self._client = HealthChecksServiceStub(grpc_channel)

    def check_health(self):
        response: CheckHealthResponse = self._client.CheckHealth(CheckHealthRequest())
        return {
            service.name: {"message": service.message, "status": service.healthy}
            for service in response.services
        }

File: packages/rust/armonik/src/client/health_checks.rs
pub use crate::rpc::health_checks::Client as HealthChecks;

use crate::client::client_method;

#[armonik_macros::client]
#[armonik(service = "armonik.api.grpc.v1.health_checks.HealthChecksService")]
impl<T: super::Channel> super::ServiceClient<crate::rpc::services::HealthChecks, T> {
    client_method!(CheckHealth:
        check()
        -> crate::health_checks::check::Request => services: Vec<crate::health_checks::ServiceHealth>);
}
